infrastructure/settings_source_factory.go

```diff
@@ -199,6 +218,10 @@ func (s *SourceOptionsSlice) UnmarshalJSON(data []byte) error {
 				var o HTTPSourceOptions
 				err, opts = mapstruc.Decode(m, &o), o
 
+			case optType == "InstanceMetadata":
```

Comment:
D'oh, totally skipped the tests.

Fixed in c94b00a8ead774c970fa449e89b8f392975ccef5